### PR TITLE
EUID Support

### PR DESCRIPTION
**Pull request by dcaunt**
- Add new API `EUIDManager` for retrieving a shared instance of `UID2Manager` configured for EUID.
- Add new `UIDSettings` property `euidEnvironment` for configuring EUID endpoints.
- Update dev app to support EUID testing via source code toggle
- Also addresses some swiftlint violation false positives

The dev app can be set to use an EUID environment by setting the `UID2EnvironmentEUID` key to `YES` in `Development/UID2SDKDevelopmentApp/UID2SDKDevelopmentApp/Info.plist` in Xcode's editor.

Alternatively you can use `plutil` from the command line:

```
plutil -replace UID2EnvironmentEUID -bool YES Development/UID2SDKDevelopmentApp/UID2SDKDevelopmentApp/Info.plist
```

to revert back to using a UID2 environment, 

```
plutil -replace UID2EnvironmentEUID -bool NO Development/UID2SDKDevelopmentApp/UID2SDKDevelopmentApp/Info.plist
```

### CODE REVIEW

File: Development/UID2SDKDevelopmentApp/UID2SDKDevelopmentApp/RootViewModel.swift [hunk 1]
```diff
@@ -31,7 +31,7 @@ class RootViewModel: ObservableObject {
         // If you have copied the dev app for testing, you probably want to use the default
         // environment, which is production.
         if Bundle.main.bundleIdentifier == "com.uid2.UID2SDKDevelopmentApp" {
-            UID2Settings.shared.environment = .custom(url: URL(string: "https://operator-integ.uidapi.com")!)
+            UID2Settings.shared.uid2Environment = .custom(url: URL(string: "https://operator-integ.uidapi.com")!)
         }
 
         return UID2Manager.shared
```
[the remark this answers is not in the document]
Comment (sunnywu) [reply Development/UID2SDKDevelopmentApp/UID2SDKDevelopmentApp/RootViewModel.swift:37]: @caroline-ttd should have provided you the EUID integ public key/subscription id credentails that should be different too

**Review by IanDBird — APPROVED** [review]

File: Sources/UID2/Networking/DataEnvelope.swift [hunk 2]
```diff
@@ -44,6 +44,8 @@ internal enum DataEnvelope {
 extension Data {
     /// A convenience initializer for converting from a Data representation of a base64 encoded string to its decoded Data.
     init?(base64EncodedData: Data, options: Data.Base64DecodingOptions = []) {
+        // https://github.com/realm/SwiftLint/issues/5263#issuecomment-2115182747
+        // swiftlint:disable:next non_optional_string_data_conversion
```
Comment (caroline-ttd) [inline Sources/UID2/Networking/DataEnvelope.swift:48]: Nitpick: `swiftlint:disable:next non_optional_string_data_conversion` what does this mean, do we need to add more explanation here

Comment (caroline-ttd) [reply Sources/UID2/Networking/DataEnvelope.swift:48]: nvm. I am new to the swift. It is used for SwiftLint to disable non_optional_string_data_conversion rule for the next line of code temporarily.

**Review by caroline-ttd — APPROVED** [review]
LGTM.

I can only use the dev app to test either EUID or UID2. I just want to confirm, it is possible for UID2 and EUID to work simultaneously, is that correct?

The primary requirement is that publisher apps should be able to create an instance of the manager to store or generate UID2 and EUID tokens based on their configuration. They should have the flexibility to use UID2, EUID, or both.

**Pull request merged**